tools/istio-iptables/pkg/capture/run.go

```diff
@@ -307,10 +307,23 @@ func ConfigureRoutes(cfg *config.Config, ext dep.Dependencies) error {
 // for IPv6 but not IPv4, as IPv4 defaults to `netmask 255.0.0.0`, which allows binding to addresses
 // in the 127.x.y.z range, while IPv6 defaults to `prefixlen 128` which allows binding only to ::1.
 // Equivalent to `ip -6 addr add "::6/128" dev lo`
-func configureIPv6Addresses(cfg *config.Config) error {
+func configureIPv6Addresses(cfg *config.Config, ext dep.Dependencies) error {
 	if !cfg.EnableInboundIPv6 {
 		return nil
 	}
+	// Issue: https://github.com/istio/istio/issues/39963
+	// netlink.LinkByName("lo") fails with "protocol not available" for Linux Kernel < 4.x.x
```

Comment:
Ah its only old kernel versions? TBH I don't feel great about adding support for 10 year old OS's with Istio alpha features (IPv6)...

If you are going to use cutting edge features on one product, expecting them to pair with ancient OS's is a bit tricky.

Reply:
We are introducing some complexity, runtime cost, and risk for all users to appease a small set of users using extremely old software

Reply:
@howardjohn RHEL 7 / Centos 7 / etc are not end of life yet and in fact would be running the majority of the production  Linux workloads. RHEL 7 only ships 3.10.x Kernels and started using 4.4.x only from RHEL8.

This issue and fix was tested on the following Kernel which was less than 2-3 months old.

`[vas@vas-istio-dev istio]$ uname -a
Linux vas-istio-dev 3.10.0-1160.66.1.el7.x86_64 #1 SMP Wed May 18 16:02:34 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
[vas@vas-istio-dev istio]$`

Reply:
@howardjohn , Additionally, the issue may not have been seen by a large userbase because this issue is seen only on IPv6 deployments.

Reply:
There are many vms running even 2.6 kernels as I know

Reply:
I agree. IMHO, we should at minimum support the current Kernels in the RHEL7/Centos7 world. (3.10.x)
I have rebased the commit and pushed again.
Can I close this review as resolved? Can you please approve if its okay? @hzxuzhonghu @howardjohn 
Thanks in advance.

Reply:
IMO we should fix upstream. This is dangerous.

How do I reproduce? I tried running centos 7, works fine...

Can you try to modify https://github.com/vishvananda/netlink/blob/afa2eb2a66aac1f8f370287f236ba93d4c078dd6/link_linux.go#L1662 to be unix.AF_INET6 and see if it works?

Reply:
@howardjohn ,
Humbly requesting to explain the dangerous part in the fix. Is there any disadvantage in using the native 'ip' command? Please help me understand if I am missing something.

IMO - native 'ip' command is robust, reliable and without compatibility issues.
Like how having IPv6 enabled in the Kernel, IP6tables modules loaded in the Kernel, etc are external dependencies...similarly, having non-trivial action which requires admin privilege like assigning IP address is an external dependency, IMO, and should be handled using native methods/commands. Please correct me and help me with an explanation if I am wrong.
Implementing a part of the capabilities of native 'ip' command internally would convert the external dependency into an internal dependency on the Kernel API/release.

For reproducing the issue, please try to inject Istio sidecar (version > 1.11 where netlink was introduced) on K8s cluster with only IPv6 enabled. Issue is seen on both istio-init and istio-cni modes.

Thanks in advance.

Reply:
`ip` command does not exist on all machines. This PR also introduces risk for newer kernels, etc if the detection logic is incorrect or slow, etc.

Reply:
1. AFAIK, `ip` command is installed even as part of the minimal installations in case of adding sidecar to VMs/baremetal/etc. If considering container images, then we require `ip` command only inside `proxyv2` images where it is already present.
2. IMO, using native `ip` command is less riskier compared to netlink in case of newer Kernels, Kernel API changes, compatability, etc.. Userbase for Linux+ip would be of substantial amount. 
3. Please let me know your concerns about the detection logic, I will try to address them. So far, I have used the `syscall` package which is used by many built-in packages such as `net` and I am calling `uname` to get the Kernel info and uname has been around for ages in Linux and do not expect this to change (AFAIK). Even if it does, Linux usually handles backward compatibilities very well given `uname` exists like forever. 
4. For latency concerns, we may try to measure as the logic is straight forward (even string operations are avoided for latency reasons).  I would like to discuss specific concerns if any so that we can address them. 

Looking forward to collaborating in taking this forward.

Reply:
```
$ docker run --entrypoint ip gcr.io/istio-testing/proxyv2:latest-distroless
docker: Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "ip": executable file not found in $PATH: unknown.
```

Distroless images (security best practice) do not have it

Reply:
ah.. okay... distroless image doesn't have them. The other proxyv2 images has it.
How was the distroless image handled in 1.11.x and before where `ip` command was used instead of netlink?

Reply:
We used a fork of distoless with `ip` added

Reply:
https://github.com/istio/istio/pull/40358#issuecomment-1244211009